src/components/NcHeaderMenu/NcHeaderMenu.vue

```diff
@@ -139,6 +140,22 @@ export default {
 			type: Boolean,
 			default: false,
 		},
+
+		/**
+		 * Pass `true` if the header menu is used for website navigation
+		 */
+		isNav: {
+			type: Boolean,
+			default: false,
+		},
+
+		/**
+		 * aria-label attribute of the wrapper element
+		 */
+		wrapperAriaLabel: {
+			type: String,
+			default: null,
+		},
```

Comment:
Do we need 2 different aria labels, one on the whole nav and one on the trigger?

Reply:
Isn't it always supposed to be the same as trigger, so we can use `aria-labledby` here?

Reply:
Yes
![image](https://github.com/nextcloud-libraries/nextcloud-vue/assets/24800714/6dda1177-2c26-4fa9-ba27-0fed63e30df4)

See https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/navigation_role#labeling_landmarks

Reply:
I still didn't understand... Could you make an example of when this component should have both `wrappedAriaLabel` and `ariaLabel`, and they are different?

Reply:
See the server draft https://github.com/nextcloud/server/pull/40168/files#diff-19af9ee8873f4260b63dd8d5974181daf4e53ab74d1d3ee84edbfd1975b215eaR27-R28

With reference to
![image](https://github.com/nextcloud-libraries/nextcloud-vue/assets/24800714/654bfbb8-f775-4980-ba65-21d5b6bb6773)
> https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/navigation_role#labeling_landmarks

and

![image](https://github.com/nextcloud-libraries/nextcloud-vue/assets/24800714/3946ede3-f912-47f8-9325-06f26853c0cb)
> https://www.unimelb.edu.au/accessibility/aria/wai-aria-landmarks-and-screen-readers

when navigating landmark regions with <kbd>d</kbd> and without unique labels, screen readers will announce them both identically as "Navigation landmark"

With unique labels users may identofy them as distinct navigation landmarks i.e. read out as "App menu (navigation landmark)" and "Settings menu (navigation landmark)"

Reply:
> With unique labels users may identify them as distinct navigation landmarks i.e. read out as "App menu (navigation landmark)" and "Settings menu (navigation landmark)"

But "App menu" and "Settings menu" will have different labels, because they are 2 different navigation - 2 different instances on `NcHeaderNav`.

Reply:
In `NcHeaderNav` we have a navigation that is opened by a button. This button controls the navigation and its label describes the navigation.

If I understand you correctly, when we use 2 labels here, we need to always manually create and control this 2 labels. And the second label is not supposed to be anything except "open" + the first label, right? Like they are always "X menu" and "Open X menu".

In this case we can use only one label on the controlling button "X menu" and make navigation `labelledby` this button.

For example, for settings menu, when a user goes to the next landmark, screen reader reads it as:
```
Settings menu navigation landmark Open settings menu button collapsed
```

But with one label and `labelled-by` it is almost the same, it is still a navigation with a button:
```
Settings menu navigation landmark Settings menu button collapsed
```

Calling button just "Settings menu" without "Open" verb seems to be to be a common practice in both Nextcloud and w3c examples (https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-links/).

In the issue example link also `aria-labelledby` is used, but with header (there is no example with both opening button and labels :( ).

Reply:
As a summarize I'm against using to an additional prop here because:
- They are always almost the same, "X menu" and "Open X Menu", but it is something on developer's responsibility, more to describe in the documentation
- One more string to translate for each nav
- It doesn't change much for a screen reader